connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -298,6 +301,50 @@ public void stopConnector(String connName) {
     }
 
     /**
+     * This method is only called when stop connector failed, connector is restarted and the starting process should be
+     * idempotent;
+     *
+     * TODO: Support tracking connector status and replace this method;
+     *
+     * @param connName
+     */
```

Comment:
Because when stopping a connector, remove it from the connectors hash map is the last step, If we catch an exception , we can assume the connector is still inside the hash map